### ISSUE TITLE
PHPCS 3.4.0: Deprecate the WordPress.WhiteSpace.CastStructureSpacing sniff

### ISSUE THREAD

**Issue by jrfnl** [WordPress/WordPress-Coding-Standards#1524]
## Rationale

PHPCS 3.4.0 will introduce a new `Generic.Formatting.SpaceBeforeCast` sniff which checks that there is exactly one space before a type cast and includes fixers for no space and too much space.

This sniff can replace the last remaining check in the `WordPress.WhiteSpace.CastStructureSpacing` sniff,

## References
* Upstream (merged) PR: https://github.com/squizlabs/PHP_CodeSniffer/pull/2217

## Action Checklist
Once PHPCS 3.4.0 has been released and WPCS is ready/willing to up the minimum required PHPCS version:
- [ ] Update the PHPCS version requirements in the `composer.json` file.
- [ ] Update references to the WPCS minimum PHPCS version requirements in the Readme and wherever else needed.
- [ ] Deprecate the `WordPress.WhiteSpace.CastStructureSpacing` sniff.
     - [ ] Make the sniff extend the upstream `Generic.Formatting.SpaceBeforeCast` sniff and turn it into an empty class.
    - [ ] Mark the sniff as deprecated.
- [ ] Once this is done, open a new issue as a reminder to remove the WP native sniff in the next major version of WPCS and to replace the ruleset reference in the `WordPress-Core` ruleset.

**Comment by jrfnl** [comment]
The upstream sniff does not make an exception for the spread operator, while based on #1762 we would want that exception. PR #1767 adds that exception to the existing WP specific sniff.

As this is quite specific, I'm not sure a similar PR would be accepted upstream. For now, this issue is superseded by PR #1767.
